py4j-java/src/main/java/py4j/Protocol.java

```diff
@@ -402,9 +402,16 @@ public static Object getPythonProxy(String commandPart, Gateway gateway) {
 			throw new Py4JException("Invalid Python Proxy.");
 		}
 
+	    ClassLoadingStrategy classLoadingStrategy = gateway.getClassLoadingStrategy();
+	    if (classLoadingStrategy == null) 
+	        classLoadingStrategy = ReflectionUtil.getClassLoadingStrategy();
+	    // Just in case
+	    if (classLoadingStrategy == null)
```

Comment:
I would remove this: by default there is a class loading strategy. If we want to prevent against a user error (setting the class loading strategy to Null, we could validate the parameter in `ReflectionUtil.setClassLoadingStrategy` instead.